src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -522,6 +522,24 @@ public Task RenderPartialAsync(string partialViewName, object model, ViewDataDic
             string templateName,
             object additionalViewData)
         {
+            if (modelExplorer.Metadata.IsEnum && modelExplorer.Model != null)
+            {
+                var value = (modelExplorer.Model as Enum).ToString("d");
+
+                var enumGrouped = modelExplorer.Metadata.EnumGroupedDisplayNamesAndValues;
+                if (enumGrouped != null &&
+                    enumGrouped.Count() > 0)
+                {
+                    foreach (var kvp in enumGrouped)
+                    {
+                        if (kvp.Value == value)
+                        {
+                            modelExplorer = modelExplorer.GetExplorerForModel(kvp.Key.Name);
```

Comment:
Deserves a comment that this will create a `ModelExplorer` with the same `Metadata` but the `Model` will be the overridden `string` representation of the `enum` value.